app/src/main/kotlin/com/mapzen/erasermap/model/InstructionGroup.kt

```diff
@@ -0,0 +1,63 @@
+package com.mapzen.erasermap.model
+
+import android.content.Context
+import com.mapzen.erasermap.R
+import com.mapzen.valhalla.Instruction
+import com.mapzen.valhalla.TravelMode
+import com.mapzen.valhalla.TravelType
+import java.util.ArrayList
+
+
+class InstructionGroup(val travelType: TravelType, val travelMode: TravelMode,
+    val instructions: ArrayList<Instruction>) {
+
+  val totalDistance: Int by lazy { calculateTotalDistance() }
+  val totalTime: Int by lazy { calculateTotalTime() }
+  var firstStationName: String? = null
+  var numberOfStops: String? = null
+
+  private fun calculateTotalDistance(): Int {
+    var total = 0
+    for(i in 0..instructions.size - 1) {
```

Comment:
Looking through the rest of the PR it seems like there's an indexing issue here that's somewhat systemic. Having to constantly `-1` the size everywhere seems odd.